### DISCUSSION TITLE
how can i run extension in intellij

### DISCUSSION THREAD

**Discussion by DasLixou** [Minestom/Minestom#1019] [Help] [answered]
how can i run extension in intellij without jaring it and putting it in server extension folder etc.

**Comment by ZakShearman** [comment, marked as answer]
Please take a read of the `Testing in a dev environment` section of the Extensions wiki: https://wiki.minestom.net/expansion/extensions#testing-in-a-dev-environment

**Reply by DasLixou** [reply]
i already did but i dont understand it. Where should i apply these and how can i do this with intelliJ?

**Reply by ZakShearman** [reply]
VM Options.

You may need to toggle the following to make VM Options show in your configuration
![](https://zak.pink/2022/05/Materialistic-Opossum-13741.png)

**Reply by DasLixou** [reply]
so i should make another class which creates an Minestom server and pass these arguements?

**Reply by ZakShearman** [reply]
You will need a class to launch your Minestom server, yes. Then just pass the arguments listed on the wiki in the VM options.

**Reply by DasLixou** [reply]
Did you put the TestServer class in main or test folder?
